modules/dxp/apps/segments/segments-experiment-test/src/testIntegration/java/com/liferay/segments/experiment/web/internal/processor/test/SegmentsExperimentSegmentsExperienceRequestProcessorTest.java

```diff
@@ -0,0 +1,457 @@
+
+		AnalyticsSyncedPrefsPropsWrapper analyticsSyncedPrefsPropsWrapper =
+			new AnalyticsSyncedPrefsPropsWrapper(prefsProps);
+
+		ReflectionTestUtil.setFieldValue(
+			PrefsPropsUtil.class, "_prefsProps",
+			analyticsSyncedPrefsPropsWrapper);
+
+		try {
+			MockHttpServletRequest mockHttpServletRequest =
+				_getMockHttpServletRequest();
+
+			mockHttpServletRequest.setParameter(
+				"segmentsExperimentKey",
+				new String[] {segmentsExperiment.getSegmentsExperimentKey()});
+
+			long[] segmentsExperienceIds =
+				_segmentsExperienceRequestProcessor.getSegmentsExperienceIds(
+					mockHttpServletRequest, new MockHttpServletResponse(),
+					_group.getGroupId(), classNameId, layout.getPlid(),
+					new long[0]);
+
+			Assert.assertEquals(
+				Arrays.toString(segmentsExperienceIds), 1,
+				segmentsExperienceIds.length);
+
+			Assert.assertEquals(
+				segmentsExperience.getSegmentsExperienceId(),
+				segmentsExperienceIds[0]);
+		}
+		finally {
+			ReflectionTestUtil.setFieldValue(
+				PrefsPropsUtil.class, "_prefsProps", prefsProps);
+		}
+	}
+
+	private Layout _addLayout() throws Exception {
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(
+				TestPropsValues.getGroupId(), TestPropsValues.getUserId());
+
+		return _layoutLocalService.addLayout(
+			TestPropsValues.getUserId(), _group.getGroupId(), false,
+			LayoutConstants.DEFAULT_PARENT_LAYOUT_ID,
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			StringPool.BLANK, LayoutConstants.TYPE_CONTENT, false,
+			StringPool.BLANK, serviceContext);
+	}
+
+	private MockHttpServletRequest _getMockHttpServletRequest()
+		throws Exception {
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		mockHttpServletRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _getThemeDisplay());
+
+		return mockHttpServletRequest;
+	}
+
+	private ThemeDisplay _getThemeDisplay() throws Exception {
+		ThemeDisplay themeDisplay = new ThemeDisplay();
+
+		themeDisplay.setCompany(
+			_companyLocalService.fetchCompany(TestPropsValues.getCompanyId()));
+		themeDisplay.setScopeGroupId(_group.getGroupId());
+		themeDisplay.setSignedIn(true);
+		themeDisplay.setUser(TestPropsValues.getUser());
+
+		return themeDisplay;
+	}
+
+	@Inject
+	private ClassNameLocalService _classNameLocalService;
+
+	@Inject
+	private CompanyLocalService _companyLocalService;
+
+	@DeleteAfterTestRun
+	private Group _group;
+
+	@Inject
+	private LayoutLocalService _layoutLocalService;
+
+	@Inject
+	private SegmentsExperienceLocalService _segmentsExperienceLocalService;
+
+	@Inject(
+		filter = "component.name=*.SegmentsExperimentSegmentsExperienceRequestProcessor"
+	)
+	private SegmentsExperienceRequestProcessor
+		_segmentsExperienceRequestProcessor;
+
+	@Inject
+	private SegmentsExperimentLocalService _segmentsExperimentLocalService;
+
+	private class AnalyticsSyncedPrefsPropsWrapper extends PrefsPropsImpl {
```

Comment:
the inner class may be static

[A non-static inner class has a reference to its outer class, and access to the outer class' fields and methods. That class reference makes the inner class larger and could cause the outer class instance to live in memory longer than necessary](https://rules.sonarsource.com/java/tag/performance/RSPEC-2694#:~:text=Code%20Smell-,Inner%20classes%20which%20do%20not,owning%20classes%20should%20be%20%22static%22&text=A%20non%2Dstatic%20inner%20class,outer%20class'%20fields%20and%20methods.&text=If%20the%20reference%20to%20the,static%20(also%20called%20nested).)

Reply:
thank you!!!, changed